pkg/webhook/webhook.go

```diff
@@ -166,7 +167,30 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) (respons
 		logger.Error("failed to marshal pod object", err)
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+
+	// patchResponse := admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+	// patches := []jsonpatch.JsonPatchOperation{}
+	// for _, patch := range patchResponse.Patches {
+	// 	// drop patches that are removing fields
+	// 	// this prevents patches that remove fields that the webhook doesn't know about
+	// 	// ref: https://github.com/Azure/azure-workload-identity/issues/1312
+	// 	if patch.Operation == "remove" {
+	// 		continue
+	// 	}
+	// 	patches = append(patches, patch)
+	// }
+	//
+	// patchResponse.Patches = patches
+	// return patchResponse
+
+	// Merge the original raw object with the modified marshaled pod
+	patchedObject, err := strategicpatch.StrategicMergePatch(req.Object.Raw, marshaledPod, corev1.Pod{})
```

Comment:
I haven't seen any other webhook use this approach, so it makes me nervous that it won't work correctly in all cases.